### PR TITLE
feat(application-system): Removing attachment field from resolver response

### PR DESCRIPTION
**Pull request by alexandradilja**
# ...

Remove attachment field from graphql queries so that the templates can not access attachment urls in aws

## What

Removed field from application model in `libs/api/domains/application/src/lib/application.model.ts`. Add and delete attachment mutations query after application id for return. 

No templates are using the attachment filed so no changes needed there. 

## Why

Security reasons.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/core/src/types/Application.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export interface ActionCardMetaData {
   }
 }
 
-export interface Application {
+export interface ApplicationModel {
```
Comment (saevarma) [inline libs/application/core/src/types/Application.ts:31]: To have this more explicit and clear in the naming how about the calling model with attachments `ApplicationWithAttachments`?

**Review by saevarma — CHANGES_REQUESTED** [review]
Looks good just question about the name of the models

File: libs/application/core/src/types/Application.ts (hunk 1 above)
Comment (alexandradilja) [reply libs/application/core/src/types/Application.ts:31]: Yes! good idea!

File: libs/application/core/src/types/Application.ts [hunk 2]
```diff
@@ -45,3 +45,5 @@ export interface Application {
   progress?: number
   status: ApplicationStatus
 }
+
+export type Application = Omit<ApplicationWithAttachments, 'attachments'>
```
Comment (miamiultras) [inline libs/application/core/src/types/Application.ts:49]: maybe the other way around and we can stick with interfaces

```
interface Application {
...
}
```

```
interface ApplicationWithAttachments extends Application {
attachments...
}
```

Comment (alexandradilja) [reply libs/application/core/src/types/Application.ts]: Yes sure!

**Review by miamiultras — APPROVED** [review]

**Review by Thorrik58 — APPROVED** [review]
LGTM

File: apps/application-system/api/src/app/modules/payment/payment.service.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ import type { User } from '@island.is/auth-nest-tools'
 import { CreateChargeResult } from './payment.type'
 import { logger } from '@island.is/logging'
 import { ApolloError } from 'apollo-server-express'
-import { Application as ApplicationModel } from '../application/application.model'
+import { Application as ApplicationWithAttachments } from '../application/application.model'
```
Comment (saevarma) [inline apps/application-system/api/src/app/modules/payment/payment.service.ts:15]: Why import the `Application` model without attachments to rename it to `ApplicationWithAttachments` ?

File: apps/financial-aid/api/src/app/modules/application/application.resolver.ts [hunk 4]
```diff
@@ -6,7 +6,7 @@ import { BackendAPI } from '../../../services'
 
 import {
   ApplicationFiltersModel,
-  ApplicationModel,
+  ApplicationWithAttachments,
```
Comment (saevarma) [inline apps/financial-aid/api/src/app/modules/application/application.resolver.ts:9]: I think you accidentally renamed the `financial-aid` related stuff?

**Review by saevarma — CHANGES_REQUESTED** [review]

File: apps/application-system/api/src/app/modules/payment/payment.service.ts (hunk 3 above)
Comment (alexandradilja) [reply apps/application-system/api/src/app/modules/payment/payment.service.ts:15]: yes, looks like it.. I'll fix that

**Review by saevarma — APPROVED** [review]
LGTM

File: libs/application/template-api-modules/src/lib/modules/templates/children-residence-change/children-residence-change.service.ts [hunk 5]
```diff
@@ -20,7 +20,7 @@ import {
   generateSyslumennNotificationEmail,
   transferRequestedEmail,
 } from './emailGenerators'
-import { Application } from '@island.is/application/core'
+import { ApplicationWithAttachments as Application } from '@island.is/application/core'
```
Comment (Rikkiabb) [inline libs/application/template-api-modules/src/lib/modules/templates/children-residence-change/children-residence-change.service.ts:23]: We are not using Attachments so we can use `Application` instead 😄

**Review by darrikonn — APPROVED** [review]

**Review by Margretol — APPROVED** [review]

**Review by johannaagma — APPROVED** [review]

**Pull request merged**